core/src/validator.rs

```diff
@@ -934,8 +934,8 @@ fn backup_and_clear_blockstore(ledger_path: &Path, start_slot: Slot, shred_versi
 
         let end_slot = last_slot.unwrap();
         info!("Purging slots {} to {}", start_slot, end_slot);
-        blockstore.purge_slots(start_slot, end_slot, PurgeType::Exact);
```

Comment:
this order was very broken, when the process is interrupted immediately after `purge_slots` and before `purge_from_next_slots`. It leaves super dangerous dangling references in meta....